builtin/common/misc_helpers.lua

```diff
@@ -458,15 +458,28 @@ do
 end
 
 --------------------------------------------------------------------------------
-function table.copy(t, seen)
-	local n = {}
-	seen = seen or {}
-	seen[t] = n
-	for k, v in pairs(t) do
-		n[(type(k) == "table" and (seen[k] or table.copy(k, seen))) or k] =
-			(type(v) == "table" and (seen[v] or table.copy(v, seen))) or v
-	end
-	return n
+
+function table.copy(t, preserve_metatables)
```

Comment:
Note: This removes / replaces the `seen` parameter, which should never have been exposed in the first place.

Reply:
IMO you can leave a warning in the changelog later if you are concerned about people using `seen`.